src/acpi-tables/src/rsdp.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use vm_memory::{Bytes, GuestAddress, GuestMemory};
+use zerocopy::little_endian::{U32, U64};
+use zerocopy::AsBytes;
+
+use crate::{checksum, Result, Sdt};
+
+/// Root System Description Pointer
+///
+/// This is the root pointer to the ACPI hierarchy. This is what OSs
+/// are looking for in the memory when initializing ACPI. It includes
+/// a pointer to XSDT
+#[repr(packed)]
+#[derive(Clone, Copy, Debug, Default, AsBytes)]
+pub struct Rsdp {
+    _signature: [u8; 8],
+    checksum: u8,
+    _oem_id: [u8; 6],
+    _revision: u8,
+    _rsdt_addr: U32,
+    _length: U32,
+    _xsdt_addr: U64,
+    extended_checksum: u8,
+    _reserved: [u8; 3],
+}
+
+impl Rsdp {
+    pub fn new(oem_id: [u8; 6], xsdt_addr: u64) -> Self {
+        let mut rsdp = Rsdp {
+            // Space in the end of string is needed!
+            _signature: *b"RSD PTR ",
+            checksum: 0,
+            _oem_id: oem_id,
+            _revision: 2,
+            _rsdt_addr: U32::ZERO,
+            _length: U32::new(std::mem::size_of::<Rsdp>().try_into().unwrap()),
+            _xsdt_addr: U64::new(xsdt_addr),
+            extended_checksum: 0,
+            _reserved: [0u8; 3],
```

Comment:
nit: no underscores on any of these

Reply:
I ignore the lint and everything is without an underscore now.